### PR TITLE
CC: Introduce secure image in payload image

### PR DESCRIPTION
**Pull request by BbolroC**
This is a preliminary work to establish an e2e test for a new runtime class `kata-qemu-se` (IBM secure execution).

Fixes: #6544

Signed-off-by: Hyounggyu Choi <Hyounggyu.Choi@ibm.com>

### CODE REVIEW

File: tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile
Comment (BbolroC) [inline tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile:1]: Rationale: got an error while trying to install a new package on a running container `build-kata-deploy` like

```
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
4383

4384
E: List directory /var/lib/apt/lists/partial is missing. - Acquire (2: No such file or directory)
4385
Reading package lists...
4386

4387
WARNING: apt does not have a stable CLI interface. Use with caution in scripts.
4388

4389
E: Could not open lock file /var/lib/dpkg/lock-frontend - open (13: Permission denied)
4390
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
```

**Comment by BbolroC** [comment]
Log for the changes: https://github.com/BbolroC/kata-containers/actions/runs/4551761798

The repository secret `CI_HKD_PATH` is configured thanks to @fidencio.

**Review by stevenhorsman — APPROVED** [review]
LGTM. Thanks!

**Review by jepio — APPROVED** [review]

File: tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile
Comment (c3d) [reply tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile:1]: I gather that the "are you root" part is misleading and that the problem is SE blocking access?

**Review by c3d — APPROVED** [review]
Looks good to me, thanks @BbolroC

File: tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile
Comment (BbolroC) [reply tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile:1]: Nope, I would say, it is rather an environmental issue for how a self-hosted runner deals with a package manager within a running container. :wink:

Comment (BbolroC) [reply tools/packaging/kata-deploy/local-build/dockerbuild/Dockerfile:1]: Disclaimer: `Dockerfile` is used to build a base image where a secure image is built. In order for that, a package `s390-tools` should be installed for a `genprotimg` program used to build a secure image. The build script was prevented from installing the package due to the permission issue above (I am sure how GHA handles this). So I decided to go for the easiest one.

**Pull request merged**